var/spack/repos/builtin/packages/expat/package.py

```diff
@@ -112,12 +112,10 @@ def url_for_version(self, version):
         url = "https://github.com/libexpat/libexpat/releases/download/R_{0}/expat-{1}.tar.bz2"
         return url.format(version.underscored, version.dotted)
 
-    def configure_args(self):
-        spec = self.spec
-        args = [
-            "--without-docbook",
-            "--enable-static",
-        ]
-        if "+libbsd" in spec and "@2.2.1:" in spec:
-            args.append("--with-libbsd")
+    def cmake_args(self):
+        args = [self.define("EXPAT_BUILD_DOCS", False), self.define("BUILD_SHARED_LIBS", False)]
```

Comment:
`BUILD_SHARED_LIBS` should probably default to `True` for both backwards-compatibility and security.

Reply:
Honestly it should probably be a variant now. W.r.t. to a default perhaps you can enlighten me to this point as I am not particularly familiar with expat's configure script but to me the previous usage of `--enable-static` would indicate the prior use of static libs no? If we're considering backwards compat at least. IMO I agree, shared is the way to go.

Reply:
Hi @johnwparent , I misread `--enable-static` as `--disable-static`, my bad, sorry.
I just checked: Autotools is building both static and shared by default, so `--enable-static` was superflous previously:
```console
# ./configure --help
[..]
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=yes]
[..]
```
CMake however builds static or shared libraries, not both, to my best knowledge, unless the upstream build system goes extra miles to do that. So if you we need both static and shared in Spack, we need to build twice some way, I guess.

Variants seem to be a lot like Gentoo USE flags but with support for multiple values to co-exist. Maybe then to have the shared-or-static flag be a variant would allow to have both at the same time where needed without building twice from *inside* the package.
So a variant and shared for a default? What do you think?

Reply:
@johnwparent PS: I haven't checked yet, but there are probably other packages around here that also had to answer the `BUILD_SHARED_LIBS` question some way.

Reply:
Right, unless the upstream project really goes out of their way (and arguably avoids best practice in doing so) CMake will only build static or shared, not both. 
The way I've been solving this as I port packages to use CMake that have been previously building both shared and static at once is to add the variant as discussed, so I'll take that approach. If a consuming package really need both the shared and static libs at once they can specify a dependency on both variants.

Reply:
@johnwparent sounds good to me :+1: